### PR TITLE
Refactor menu actions and shortcut handling

### LINKED ISSUE #3074: Unexpected error: Invalid state: all arguments must be non-null.
### Description

An unexpected error occurred in the main process.
This error occurred after changing the keyboard bindings, specifically changing the ctrl+1,2,3,4,5,6 to set heading levels.
After saving the settings changed, I immediately closed the window.
simply reinstalling (without uninstalling first) did not fix the issue and it remains unresolved. (haven't tried anything else yet)
Thank you.

### Stack Trace

```
Error: Invalid state: all arguments must be non-null.
    at Object.register (C:\Program Files\Mark Text\MarkText\resources\app.asar\node_modules\@hfelix\electron-localshortcut\src\electron-localshortcut.js:167:11)
    at Object.registerKeyHandlers (C:\Program Files\Mark Text\MarkText\resources\app.asar\dist\electron\main.js:2:288839)
    at Object.addEditorMenu (C:\Program Files\Mark Text\MarkText\resources\app.asar\dist\electron\main.js:2:295113)
    at Xt.createWindow (C:\Program Files\Mark Text\MarkText\resources\app.asar\dist\electron\main.js:2:254198)
    at Object._createEditorWindow (C:\Program Files\Mark Text\MarkText\resources\app.asar\dist\electron\main.js:2:302982)
    at App.<anonymous> (C:\Program Files\Mark Text\MarkText\resources\app.asar\dist\electron\main.js:2:300927)
    at App.emit (node:events:394:28)
```

### Version

MarkText: v0.17.0
Operating system: Windows_NT x64 10.0.22000 (win32)

### PR DESCRIPTION
**Pull request by fxha**
| Q                 | A
| ----------------- | ---
| License           | MIT

### Description

Refactored menu actions and shortcut handling by adding a command manager to the main process. This will improve the overall code quality and enables us to change key binding at runtime.

----

fixes #3032
fixes #3074

### CODE REVIEW

**Comment by Jocs** [comment]
@fxha Please fix the conflict.

**Review by Jocs — APPROVED** [review]

**Comment by Jocs** [comment]
>  change key binding at runtime.

1. It seems not work, when I use `cmd+[` to copy some text and it not work after I change `cmd+c` to `cmd+[`.
2. The shortcut in the menu is not changed to `cmd+[` after I change `cmd+c` to `cmd+[`.

File: docs/KEYBINDINGS_OSX.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ MarkText key bindings for macOS. Please see [general key bindings](KEYBINDINGS.m
 
 | Id                  | Default                                          | Description                           |
 |:------------------- | ------------------------------------------------ | ------------------------------------- |
-| `file.new-file`     | <kbd>Command</kbd>+<kbd>N</kbd>                  | New file                              |
+| `file.new-window`   | <kbd>Command</kbd>+<kbd>N</kbd>                  | New file                              |
```
[the remark this answers is not in the document]
Comment (fxha) [reply docs/KEYBINDINGS_OSX.md]: It's was previously new file but we changed it sometime to new window and missed updating the ID and documentation.

**Comment by fxha** [comment]
> > change key binding at runtime.
> 
> @fxha I found two issues:
> 
>     1. It seems not work, when I use `cmd+[` to copy some text after I change `cmd+c` to `cmd+[`.
> 
>     2. The shortcut in the menu is not changed to `cmd+[` after I change `cmd+c` to `cmd+[`.

Changing key bindings at runtime isn't yet implemented but is planned in a future PR.

**Pull request merged**